plugin-cgroupv2/src/k8s/token.rs

```diff
@@ -55,70 +55,65 @@ impl Token {
     /// Returns the current value of the token if it is still alive or
     /// it's refreshed value if not.
     pub async fn get_value(&self) -> anyhow::Result<String> {
-        if self.is_valid().await {
-            return match &self.data.read().await.value {
-                Some(value) => Ok(value.clone()),
-                None => Err(anyhow!("could no read the token's value")),
-            };
+        match self.is_valid().await {
+            true => {
+                let data_locked = self.data.read().await;
+                match &data_locked.value {
+                    Some(value) => Ok(value.clone()),
+                    None => Err(anyhow!("ERROR : Could not read the token's value")),
+                }
+            }
+            false => match self.refresh().await {
+                Ok(token) => Ok(token),
+                Err(e) => Err(anyhow!("ERROR : Failed to refresh token: {}", e)),
+            },
         }
-
-        self.refresh().await
     }
 
     /// Retrieves the k8s API token using either a kubectl command
     /// or by reading  the service account token's file.
     async fn refresh(&self) -> anyhow::Result<String> {
-        let token = match self.retrieval {
+        let token: String = match self.retrieval {
             TokenRetrieval::Kubectl => {
-                let output = Command::new("kubectl")
+                let output: std::process::Output = Command::new("kubectl")
                     .args(["create", "token", "alumet-reader", "-n", "alumet"])
                     .output()
-                    .map_err(|e| anyhow!("Failed to execute kubectl command: {}", e))?;
+                    .context("ERROR : kubectl command execution")?;
 
-                if !output.status.success() {
-                    return Err(anyhow!(
-                        "kubectl raised an error: {}",
-                        String::from_utf8_lossy(&output.stderr)
-                    ));
-                }
-
-                String::from_utf8_lossy(&output.stdout).trim().to_string()
+                String::from_utf8(output.stdout)
+                    .context("ERROR : UTF-8 output conversion")?
+                    .trim()
+                    .to_string()
             }
+
             TokenRetrieval::File => {
                 let mut file = File::open("/var/run/secrets/kubernetes.io/serviceaccount/token")
-                    .map_err(|e| anyhow!("Failed to open service account token file: {}", e))?;
+                    .context("ERROR : opening token file")?;
                 let mut token = String::new();
-                file.read_to_string(&mut token)
-                    .map_err(|e| anyhow!("Failed to read service account token file: {}", e))?;
+                file.read_to_string(&mut token).expect("ERROR : Reading token file");
```

Comment:
You should probably use `context` and `?` here.
Please read https://doc.rust-lang.org/book/ch09-02-recoverable-errors-with-result.html